src/components/TaskList.tsx

```diff
@@ -22,12 +22,28 @@ export default TaskList;
 
 function ExpandableTaskItem({ item }: { item: TaskItem }) {
   const [isExpanded, setIsExpanded] = useState(false);
+  const [isChecked, setIsChecked] = useState(false);
 
   // function which will not be re-created for each re-rendering
   const handleClick = useCallback(
     () => setIsExpanded((expanded) => !expanded),
     []
   );
+  const handleCheckboxChange = useCallback(() => {
+    setIsChecked((checked) => !checked);
+  }, []);
+
+  useEffect(() => {
+    const checkbox = window.localStorage.getItem(`TASK_CHECKED:${item.id}`);
```

Comment:
I think you dont need to use the `window` object for this
```suggestion
    const checkbox = localStorage.getItem(`TASK_CHECKED:${item.id}`);
```

Reply:
ALso, `TASK_CHECKED` its being used twice. WDYT of moving it to a constant variable and reusing it?